bundles/org.pitest.pitclipse.ui/src/org/pitest/pitclipse/ui/view/PitView.java

```diff
@@ -53,9 +98,57 @@ public void setFocus() {
     @Override
     public synchronized void update(File result) {
         if (result == null) {
+            homeUrlString = null;
             browser.setText("<html/>");
         } else {
-            browser.setUrl(result.toURI().toString());
+            homeUrlString = result.toURI().toString();
+            // fallback, if the file does not exist
+            browser.setUrl(homeUrlString);
+            String url = getUrlToLastActiveClass(result);
+            if (url != null && new File(url).exists()) {
+                browser.setUrl(url);
+            }
         }
     }
+
+    /**
+     * Looks up the last active file from an editor and constructs a path to its
+     * page of the result
+     * 
+     * @param result file which is used for the path beginning
+     * @return String which represents the path in the result folder or null, if the
+     *         file form the editor is no .java file or no FileEditor was active
+     *         last
+     */
+    private String getUrlToLastActiveClass(File result) {
+        IEditorInput input = getSite().getPage().getActiveEditor().getEditorInput();
+        if (input instanceof FileEditorInput) {
+            final IFile file = ((FileEditorInput) input).getFile();
+            return getFilePathInResultDir(file.getFullPath().toString(), result.getParent());
+        }
+        return null;
+    }
+
+    /**
+     * Gets the path of the given path in context with the base dir of an result
+     * 
+     * @param filePath of file, which should be present in pit result
+     * @param baseDir  of the results in HTML format
+     * @return file path in context of the base dir
+     */
+    public static String getFilePathInResultDir(String filePath, String baseDir) {
+        // replace separator chars with dots for regEx
+        filePath = filePath.replace(File.separatorChar, '.');
+        Pattern pattern = Pattern.compile("(.*\\.src\\.)(.*)(\\.)(.*\\.java)");
+        Matcher matcher = pattern.matcher(filePath);
+        if (matcher.find() && matcher.groupCount() == 4) {
+            // group 1 and 3 are not from interest
+            return baseDir + File.separatorChar + matcher.group(2) + File.separatorChar + matcher.group(4) + ".html";
+        }
+        return null;
+    }
+
+    private boolean isBrowserSet() {
+        return browser != null && homeUrlString != null;
```

Comment:
how can the browser be null?

Reply:
If in createPartControl() an exception is thrown while creating the browser.

Reply:
Well in that case we have more serious problems anyway, so it's useless to check for the browser to be null ;)
even if the homeUrlString is null it should be safe to go back or forward in the browser, shouldn't it? So this check might be useless